### PR TITLE
feat: added implementation to reorder mosaic layers

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                       |
| Breaking change |✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                      |
| Chore            | ✔                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: src/layers/models/layersManager.ts [hunk 1]
```diff
@@ -75,9 +76,32 @@ export class LayersManager {
...
 
     replaceYamlFileContent(this.mapproxyConfig.yamlFilePath, yamlContent);
     this.logger.log('info', `Successfully added layer: '${layerToMosaicRequest.layerName}' to mosaic: '${layerToMosaicRequest.mosaicName}'`);
   }
+
+  public reorderMosaic(reorderMosaicRequest: IReorderMosaicRequest): void {
+    this.logger.log('info', `Reorder mosaic: ${reorderMosaicRequest.mosaicName} request`);
+    const jsonDocument: IMapProxyJsonDocument = convertYamlToJson(this.mapproxyConfig.yamlFilePath);
+    if (!isLayerNameExists(jsonDocument, reorderMosaicRequest.mosaicName)) {
+      throw new NoContentError(`Mosaic name '${reorderMosaicRequest.mosaicName}' is not exists`);
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:89]: NoContentError - consider using 404

File: src/layers/models/layersManager.ts [hunk 2]
```diff
@@ -75,9 +76,32 @@ export class LayersManager {
...
+    }
+    reorderMosaicRequest.layers.forEach((layer) => {
+      if (!isLayerNameExists(jsonDocument, layer.layerName)) {
+        throw new NoContentError(`layer name '${layer.layerName}' is not exists`);
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:93]: NoContentError - consider using 404

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: src/layers/models/layersManager.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts]: fixed in ceb8c7c

File: src/layers/models/layersManager.ts (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts]: fixed in ceb8c7c

File: src/common/utils.ts [hunk 3]
```diff
@@ -32,3 +32,17 @@ export function replaceYamlFileContent(yamlFilePath: string, yamlContent: string
     throw new Error(error);
   }
 }
+
+// sort an array in numerical order
+export function sortArrayByZIndex(layersArr: IReorderMosaicLayerObject[]): string[] {
+  try {
+    const sortedArray = layersArr.sort((a, b) => a.zIndex - b.zIndex);
+    const layers: string[] = [];
+    sortedArray.forEach((layer) => {
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply src/common/utils.ts]: fixed in ceb8c7c

File: src/layers/models/layersManager.ts [hunk 4]
```diff
@@ -63,21 +64,45 @@ export class LayersManager {
   public addLayerToMosaic(layerToMosaicRequest: ILayerToMosaicRequest): void {
     this.logger.log('info', `Add layer: ${layerToMosaicRequest.layerName} to mosaic: ${layerToMosaicRequest.mosaicName} request`);
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const jsonDocument: IMapProxyJsonDocument | undefined = convertYamlToJson(this.mapproxyConfig.yamlFilePath);
+    const jsonDocument: IMapProxyJsonDocument = convertYamlToJson(this.mapproxyConfig.yamlFilePath);
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:67]: add all error types to openapi

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/layers/models/layersManager.ts (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts:67]: fixed in e8b20a3

**Review by asafMasa — APPROVED** [review]

**Pull request merged**